diff -r 10a002640754 v4l2-apps/lib/libv4l/libv4l1/Makefile
--- a/libv4l1/Makefile	Mon Sep 15 13:48:21 2008 +0200
+++ b/libv4l1/Makefile	Thu Oct 02 18:45:27 2008 +0200
@@ -3,7 +3,7 @@ CFLAGS := -g -O1
 CFLAGS := -g -O1
 CFLAGS += -Wall -Wno-unused -Wpointer-arith -Wstrict-prototypes -Wmissing-prototypes
 
-LIBS = -lpthread
+LIBS_libv4l1  = -lpthread
 
 V4L1_OBJS     = libv4l1.o log.o
 V4L1COMPAT    = v4l1compat.so
@@ -75,7 +75,7 @@ clean::
 	$(CC) -c -MMD $(CPPFLAGS) $(CFLAGS) -o $@ $<
 
 %.so:
-	$(CC) -shared $(LDFLAGS) -Wl,-soname,$@.$(LIB_RELEASE) -o $@.$(LIB_RELEASE) $^ $(LIBS)
+	$(CC) -shared $(LDFLAGS) -Wl,-soname,$@.$(LIB_RELEASE) -o $@.$(LIB_RELEASE) $^ $(LIBS_$*)
 	ln -f -s $@.$(LIB_RELEASE) $@
 
 %.a:
diff -r 10a002640754 v4l2-apps/lib/libv4l/libv4l2/Makefile
--- a/libv4l2/Makefile	Mon Sep 15 13:48:21 2008 +0200
+++ b/libv4l2/Makefile	Thu Oct 02 18:44:24 2008 +0200
@@ -3,7 +3,7 @@ CFLAGS := -g -O1
 CFLAGS := -g -O1
 CFLAGS += -Wall -Wno-unused -Wpointer-arith -Wstrict-prototypes -Wmissing-prototypes
 
-LIBS = -lpthread
+LIBS_libv4l2  = -lpthread
 
 V4L2_OBJS     = libv4l2.o log.o
 V4L2CONVERT   = v4l2convert.so
@@ -74,7 +74,7 @@ clean::
 	$(CC) -c -MMD $(CPPFLAGS) $(CFLAGS) -o $@ $<
 
 %.so:
-	$(CC) -shared $(LDFLAGS) -Wl,-soname,$@.$(LIB_RELEASE) -o $@.$(LIB_RELEASE) $^ $(LIBS)
+	$(CC) -shared $(LDFLAGS) -Wl,-soname,$@.$(LIB_RELEASE) -o $@.$(LIB_RELEASE) $^ $(LIBS_$*)
 	ln -f -s $@.$(LIB_RELEASE) $@
 
 %.a:
