Author: Robert Millan <rmh@debian.org>
Description: Add kFreeBSD support
Bug-Debian: http://bugs.debian.org/647508

--- a/include/linux/videodev2.h
+++ b/include/linux/videodev2.h
@@ -61,9 +61,28 @@
 #else
 #include <sys/time.h>
 #endif
+
+#ifdef linux
 #include <linux/compiler.h>
 #include <linux/ioctl.h>
 #include <linux/types.h>
+#elif defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
+#include <sys/ioctl.h>
+#include <stdint.h>
+typedef int8_t __s8;
+typedef int16_t __s16;
+typedef int32_t __s32;
+typedef int64_t __s64;
+typedef uint8_t __u8;
+typedef uint16_t __u16;
+typedef uint32_t __u32;
+typedef uint64_t __u64;
+typedef uint16_t __le16;
+typedef uint32_t __le32;
+typedef uint64_t __le64;
+#define __user
+#endif
+
 #include <linux/v4l2-common.h>
 #include <linux/v4l2-controls.h>
 
--- a/lib/include/libv4l1-videodev.h
+++ b/lib/include/libv4l1-videodev.h
@@ -2,7 +2,12 @@
 #ifndef __LINUX_VIDEODEV_H
 #define __LINUX_VIDEODEV_H
 
+#ifdef linux
 #include <linux/ioctl.h>
+#elif defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
+#include <sys/ioctl.h>
+#endif
+
 #include <stdint.h>
 
 #define VID_TYPE_CAPTURE	1	/* Can capture */
--- a/lib/include/libv4lconvert.h
+++ b/lib/include/libv4lconvert.h
@@ -26,9 +26,7 @@
 #include <sys/time.h>
 #include <linux/types.h>
 #include <linux/ioctl.h>
-#endif
-
-#ifdef __FreeBSD__
+#elif defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
 #include <sys/time.h>
 #include <sys/types.h>
 #include <sys/ioctl.h>
--- a/lib/libv4lconvert/libv4lsyscall-priv.h
+++ b/lib/libv4lconvert/libv4lsyscall-priv.h
@@ -47,9 +47,7 @@
 #define	SYS_mmap2 SYS_mmap
 #define	MMAP2_PAGE_SHIFT 0
 #endif
-#endif
-
-#ifdef __FreeBSD__
+#elif defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
 #include <sys/time.h>
 #include <sys/syscall.h>
 #include <sys/types.h>
@@ -83,9 +81,9 @@
 #define SYS_WRITE(fd, buf, len) \
 	syscall(SYS_write, (int)(fd), (void *)(buf), (size_t)(len));
 
-#ifdef __FreeBSD__
+#if defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
 #define SYS_MMAP(addr, len, prot, flags, fd, off) \
-	__syscall(SYS_mmap, (void *)(addr), (size_t)(len), \
+	syscall(SYS_mmap, (void *)(addr), (size_t)(len), \
 			(int)(prot), (int)(flags), (int)(fd), (__off_t)(off))
 #else
 #define SYS_MMAP(addr, len, prot, flags, fd, off) \
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,10 @@
+UNAME = $(shell uname -s)
+
 all install:
 	$(MAKE) -C lib $@
+ifeq ($(UNAME), Linux)
 	$(MAKE) -C utils $@
+endif
 
 sync-with-kernel:
 	@if [ ! -f $(KERNEL_DIR)/include/linux/videodev2.h -o \
