Author: Gregor Jasny <gjasny@googlemail.com>
Origin: upstream, http://git.linuxtv.org/cgit.cgi/v4l-utils.git/commit/?id=0f2adbbfe1a5bf42fad74e6087fd2075caed7cfd
Description: Remove obsolete include
Forwarded: not-needed
--- a/lib/libdvbv5/dvb-file.c
+++ b/lib/libdvbv5/dvb-file.c
@@ -43,7 +43,6 @@
 #include <libdvbv5/desc_sat.h>
 #include <libdvbv5/desc_terrestrial_delivery.h>
 #include <libdvbv5/desc_service.h>
-#include <libdvbv5/desc_service_list.h>
 #include <libdvbv5/desc_frequency_list.h>
 #include <libdvbv5/desc_event_short.h>
 #include <libdvbv5/desc_event_extended.h>
