--- a/Make.rules
+++ b/Make.rules
@@ -11,6 +11,7 @@
 LIBDIR = $(PREFIX)/lib
 # subdir below LIBDIR in which to install the libv4lx libc wrappers
 LIBSUBDIR = libv4l
+LIBCONVERTSUBDIR = $(LIBSUBDIR)
 MANDIR = $(PREFIX)/share/man
 
 # These ones should not be overriden from the cmdline
--- a/lib/libv4lconvert/Makefile
+++ b/lib/libv4lconvert/Makefile
@@ -26,7 +26,7 @@
 ONVERT_OBJS += jpeg_memsrcdest.o
 endif
 
-override CPPFLAGS += -DLIBDIR=\"$(LIBDIR)\" -DLIBSUBDIR=\"$(LIBSUBDIR)\"
+override CPPFLAGS += -DLIBDIR=\"$(LIBDIR)\" -DLIBSUBDIR=\"$(LIBCONVERTSUBDIR)\"
 
 all: $(TARGETS)
 
@@ -48,7 +48,7 @@
 install: all
 	mkdir -p $(DESTDIR)$(PREFIX)/include
 	install -p -m 644 $(INCLUDES) $(DESTDIR)$(PREFIX)/include
-	mkdir -p $(DESTDIR)$(LIBDIR)/$(LIBSUBDIR)
+	mkdir -p $(DESTDIR)$(LIBDIR)/$(LIBCONVERTSUBDIR)
 ifeq ($(LINKTYPE),static)
 	install -m 644 $(CONVERT_LIB) $(DESTDIR)$(LIBDIR)
 else
@@ -56,7 +56,7 @@
 	cd $(DESTDIR)$(LIBDIR) && \
 	  ln -f -s $(CONVERT_LIB).$(LIB_RELEASE) $(CONVERT_LIB)
 endif
-	install -m 755 *-decomp $(DESTDIR)$(LIBDIR)/$(LIBSUBDIR)
+	install -m 755 *-decomp $(DESTDIR)$(LIBDIR)/$(LIBCONVERTSUBDIR)
 	mkdir -p $(DESTDIR)$(LIBDIR)/pkgconfig
 	install -m 644 libv4lconvert.pc $(DESTDIR)$(LIBDIR)/pkgconfig
 
