From: Gregor Jasny <gjasny@googlemail.com>
Date: Sat, 19 Jan 2013 17:22:53 +0100
Subject: [PATCH] libv4l: Add ASUS F2F with STK DC-1125 to upside down list
Origin: upstream, http://git.linuxtv.org/v4l-utils.git/patch/da77f2e123d85bd03a69c4a35dbdefa2f26b047b

--- a/lib/libv4lconvert/control/libv4lcontrol.c
+++ b/lib/libv4lconvert/control/libv4lcontrol.c
@@ -268,7 +268,7 @@
 static const char *asus_board_name[] = {
 	"A3[A-Z]*",
 	"B50[A-Z]*",
-	"F[3579][A-Z]*", "F70[A-Z]*", "F[58]2[A-Z]*",
+	"F[23579][A-Z]*", "F70[A-Z]*", "F[58]2[A-Z]*",
 	"G[12][A-Z]*", "G[57]0[A-Z]*",
 	"K[4567]0[A-Z]*", "K[56]1[A-Z]*", "K52[A-Z]*", "K[45]3[A-Z]*",
 	"N[12579]0[A-Z]*", "N[56]1[A-Z]*", "N82[A-Z]*", "N[47]3[A-Z]*",
@@ -314,6 +314,7 @@
 	{ 0x174f, 0x8a31 },
 	{ 0x174f, 0xa311 },
 	{ 0x1d4d, 0x1002 },
+	{ 0x05e1, 0x0501 },
 	{ 0x0000, 0x0000 }
 };
 
